### PR TITLE
Fixed JmacHttpsTest - certificate authentication

### PR DESCRIPTION
**Pull request by dmatej**
- created DescriptorPrincipalName class
- Note that the equals method is not symmetric here
- Exousia uses Set.contains -> equals to check the authorization.
- I had to improve logs and code on several places to find out what is happening, later I partially separated that to standalone commit.

### CODE REVIEW

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java [hunk 1]
```diff
@@ -524,8 +501,8 @@ public java.lang.Object run() {
                 });
             }
         } catch (Exception ex) {
-            if (_logger.isLoggable(Level.FINE)) {
-                _logger.log(Level.FINE, "Exception when trying to populate groups for CallerPrincipal " + identityAssertion, ex);
+            if (_logger.isLoggable(FINE)) {
```
Comment (avpinchuk) [inline nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java:504]: here too

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java [hunk 2]
```diff
@@ -537,8 +514,8 @@ public java.lang.Object run() {
      *
      */
     private static void doCertificateLogin(Subject s) throws LoginException {
-        if (_logger.isLoggable(Level.FINE)) {
-            _logger.log(Level.FINE, "Processing X509 certificate login.");
+        if (_logger.isLoggable(FINE)) {
```
Comment (avpinchuk) [inline nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java:517]: Same

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java [hunk 3]
```diff
@@ -650,7 +619,7 @@ public static void doX500Login(Subject s, String appModuleID) throws LoginExcept
                 setSecurityContext(user, s, realm_name);
             }
 
-            if (_logger.isLoggable(Level.FINE)) {
+            if (_logger.isLoggable(FINE)) {
```
Comment (avpinchuk) [inline nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java:622]: same

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java [hunk 4]
```diff
@@ -307,17 +303,16 @@ private static void doPasswordLogin(Subject subject) throws LoginException {
...
+            throw new LoginException("Failed obtaining the JAAS Context", ex);
         }
 
         assert user != null;
         assert pwd != null;
         assert realm != null;
         assert jaasCtx != null;
 
-        if (_logger.isLoggable(Level.FINE)) {
+        if (_logger.isLoggable(FINE)) {
```
[the remark this answers is not in the document]
Comment (dmatej) [reply nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/login/LoginContextDriver.java:315]: Eh, yeah, these changes were done by Eclipse, just static import, but you are right, I should fix that too when I already touched it.

**Comment by dmatej** [comment]
> Local `mvn clean install -Pstaging` of your branch successful.
> 
> Perhaps some problems with k8s.

Hmmmm ...
```
22:10:54  io.fabric8.kubernetes.client.http.WebSocketHandshakeException
22:10:54  	at io.fabric8.kubernetes.client.http.StandardHttpClient.lambda$buildWebSocket$17(StandardHttpClient.java:243)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)
22:10:54  	at io.fabric8.kubernetes.client.utils.AsyncUtils.lambda$retryWithExponentialBackoff$3(AsyncUtils.java:90)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
22:10:54  	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)
22:10:54  	at io.fabric8.kubernetes.client.okhttp.OkHttpWebSocketImpl$1.onFailure(OkHttpWebSocketImpl.java:88)
22:10:54  	at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.kt:592)
22:10:54  	at okhttp3.internal.ws.RealWebSocket$connect$1.onResponse(RealWebSocket.kt:174)
22:10:54  	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
22:10:54  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
22:10:54  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
22:10:54  	at java.base/java.lang.Thread.run(Thread.java:839)
22:10:54  Caused by: java.net.ProtocolException: Expected HTTP 101 response but was '500 Internal Server Error'
22:10:54  	at okhttp3.internal.ws.RealWebSocket.checkUpgradeSuccess$okhttp(RealWebSocket.kt:224)
22:10:54  	at okhttp3.internal.ws.RealWebSocket$connect$1.onResponse(RealWebSocket.kt:170)
22:10:54  	... 4 more
```

**Comment by avpinchuk** [comment]
Yes, same error on other branch's build...

**Comment by dmatej** [comment]
The problem was reported to GitLab: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4475

**Comment by dmatej** [comment]
Interesting how that HTTP500 is reproducible just on the Jenkins. Seems I hit some sensitive place ...

**Comment by dmatej** [comment]
Meh, aha. I got it. The file is the same file in domain lib dir, test deleted it, another test created the same file with different content, but as the domain did not restart and classes are still cached by some classloader, boooom!

Such external library is owned by the domain, not by the application, so basically it is a correct behavior, because the test "corrupted" domain's files. 

Possible solutions:
* stop the domain before removing/changing its libraries
* use unique names for different libraries
* include the library to the war file (I am not sure if it would work here, but I can try).

**Review by avpinchuk — APPROVED** [review]

**Pull request merged**